.github/workflows/issue-to-pr.yaml

```diff
@@ -0,0 +1,189 @@
+name: Issue Submission to Pull Request
+on:
+  issues:
+    types:
+      [opened, edited]
+
+jobs:
+  provider_issue_to_pr:
+    if: contains(github.event.issue.labels.*.name, 'provider') && contains(github.event.issue.labels.*.name, 'submission') 
+    runs-on: ubuntu-latest
+    permissions:
+      issues: write
+      contents: write
+      pull-requests: write
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version-file: './src/go.mod'
+      - name: Provider Issue to Pull Request
+        run: |
+          # Pull provider out of issue body
+          provider=$(echo "$BODY" | grep "### Repository" -A2 | tail -n1 | sed -e 's/[\r\n]//g')
+          echo "Detected Provider '$provider'"
+
+          # Check that provider matches expected pattern
+          if ! [[ "$provider" =~ [a-zA-Z0-9\-]+/terraform-provider-[a-zA-Z0-9\-]+ ]]; then
```

Comment:
This validation is probably good enough for now, but we might want to harden it further, to further align with the CLI.

For example, [it doesn't allow for multiple dashes in a row](https://github.com/opentofu/registry-address/blob/901b9ae4061a6e83e7db96e71b8d7088294930b3/provider.go#L422-L429)

Reply:
Pulled in that dependency.